tests/test_validate.py

```diff
@@ -0,0 +1,102 @@
+"""
+Tests on functions in common.
+"""
+
+import unittest
+from logging import getLogger
+from distutils.version import StrictVersion
+from pandas import DataFrame
+import sklearn
+from skl2onnx.validate import (
+    enumerate_validated_operator_opsets, sklearn_operators,
+    summary_report
+)
+import onnx
+
+
+def has_openpyxl():
+    try:
+        import pydot  # noqa
+        return True
+    except ImportError:
+        return False
+
+
+class TestValidate(unittest.TestCase):
+
+    def test_sklearn_operators(self):
+        res = sklearn_operators()
+        assert len(res) > 0
+        assert len(res[0]) == 3
+
+    @unittest.skipIf(StrictVersion(onnx.__version__) < StrictVersion("1.4.0"),
+                     reason="OnnxOperator not working")
+    @unittest.skipIf(("dev" not in sklearn.__version__ and
+                      StrictVersion(sklearn.__version__) <
+                      StrictVersion("0.21")),
+                     reason="needed only once")
+    @unittest.skipIf(not has_openpyxl(), reason="missing dependency")
```

Comment:
Given you already put openyxl in the requirements-dev.txt, please remove this skip line.
Too many skip lines will make other fully be lost about how to run the full test.

Reply:
Wrong copy paste, it checks pydot, py dot is part of the requirements, but it still needs graphviz which not necessarily installed.